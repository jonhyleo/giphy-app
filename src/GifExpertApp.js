import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";
import Header from "./components/Header";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  return (
    <>
      <Header />

      <section className="container">
        <AddCategory setCategories={setCategories} />
        <hr />

        <div className="row">
          {categories.map((category, i) => (
            <GifGrid category={category} key={i} />
          ))}
        </div>
      </section>
    </>
  );
};

export default GifExpertApp;
