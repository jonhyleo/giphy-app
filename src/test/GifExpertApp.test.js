import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Test on GifExpertApp", () => {
  test("should render component properly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show a categories list", () => {
    const categories = ["Dragon Ball", "One Punch"];
    const wrapper = shallow(<GifExpertApp categories={categories} />);

    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
