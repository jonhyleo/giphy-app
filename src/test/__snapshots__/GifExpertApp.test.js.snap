// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifExpertApp should render component properly 1`] = `
<Fragment>
  <Header />
  <section
    className="container"
  >
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div
      className="row"
    />
  </section>
</Fragment>
`;

exports[`Test on GifExpertApp should show a categories list 1`] = `
<Fragment>
  <Header />
  <section
    className="container"
  >
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div
      className="row"
    />
  </section>
</Fragment>
`;
